### PR TITLE
Fix LLVM 3.7 build

### PR DESCRIPTION
**Pull request by Keno**

### CODE REVIEW

File: src/debuginfo.cpp [hunk 1]
```diff
@@ -68,7 +68,7 @@ struct FuncInfo {
 #else
 struct ObjectInfo {
     const object::ObjectFile* object;
-    size_t size;
+    size_t SectionSize;
```
Comment (vtjnash) [inline src/debuginfo.cpp]: size is supposed to be SymbolSize, not SectionSize. this change could occasional corrupt the backtrace structures.

File: src/debuginfo.cpp [hunk 2]
```diff
@@ -219,13 +219,11 @@ class JuliaJITEventListener: public JITEventListener
 #endif
 #endif
 
-#ifdef _OS_WINDOWS_
 #ifndef LLVM36
         uint64_t SectionAddr = 0;
 #endif
         uint64_t SectionSize = 0;
         uint64_t SectionAddrCheck = 0; // assert that all of the Sections are at the same location
```
Comment (vtjnash) [inline src/debuginfo.cpp]: this variable is only used on windows, and needs the ifdef test

File: src/debuginfo.cpp [hunk 3]
```diff
@@ -330,11 +329,11 @@ class JuliaJITEventListener: public JITEventListener
             if (!Addr) continue;
 #   endif
 #elif defined(_OS_WINDOWS_)
-#   if defined(LLVM36)
-            SectionSize = Section->getSize();
+#   if defined(LLVM37)
+            assert(obj.isELF)
```
Comment (vtjnash) [inline src/debuginfo.cpp]: this assert seems to be incorrect. COFF also has this method (http://llvm.org/docs/doxygen/html/COFFObjectFile_8cpp_source.html#l00239)

File: src/debuginfo.cpp (hunk 1 above)
Comment (Keno) [inline src/debuginfo.cpp]: SymbolSize is not a meaningful quantity for most object formats.

File: src/debuginfo.cpp (hunk 3 above)
Comment (Keno) [inline src/debuginfo.cpp]: Only for common symbols.

**Pull request merged**